### PR TITLE
First attempt to implement functionality of issue #33

### PR DESCRIPTION
**Pull request by adpe**
Hi @martignoni 

That's my first attempt. It includes a commit with a fix for Grunt.

I have some open questions and ideas to implement this nicer. What about a `modal` where you must confirm with yes/no, to proceed with reboot/shutdown?

Furthermore, I'm passing now the argument in the URL. Maybe there's a better solution.

Greets
Adrian

### CODE REVIEW

**Comment by martignoni** [comment]
Thanks @adpe. I'll review this ASAP (being very busy now).

**Comment by martignoni** [comment]
I didn't forget this, but did not have the time to review it.

**Comment by martignoni** [comment]
Finally have had to review this. Many, many thanks! 🙇 This is really a very good start! 👍

Some notes:

- Instead of being in `lib.php`, the new helper function `tool_moodlebox_standard_footer_html()` should be placed into `classes/local/utils.php`.
- We should use the [Moodle form API](https://docs.moodle.org/dev/Form_API) to display the buttons. This will make the way of triggering the actions using `$_GET['init']` not necessary.

When testing, the buttons did work (the URL changed), but the trigger files would not be modified, and consequently the MoodleBox would not restart, nor shutdown. I did not investigate why, because I imagine that if we use the Moodle form API, this won't be a problem anymore.

Last note: this will need a preference setting, so that the MoodleBox admin can either switch on/off the feature.

**Comment by adpe** [comment]
I'll work on it in the next few days.

**Comment by adpe** [comment]
We must kept this in the lib.php, because this function will be called so on each Moodle site request. How do we want that Moodle knows to call this function on each request, when it's located in the `utils` lib?

I've now reused your form with the buttons, and created a new form to handle the preference setting of this functionality.

I think the solution right now is not bad. Please let me know if you see possibilities to improve that. And feel free to adapt the strings.

File: index.php [hunk 1]
```diff
@@ -325,9 +325,28 @@
             $OUTPUT->help_icon('restartstop', 'tool_moodlebox'), 'moodleboxrestartstopsection');
     echo $OUTPUT->box_start('generalbox');
 
+    $restartstopfilename = '.restartstopsettings';
```
Comment (martignoni) [inline index.php]: This would be better done via an admin setting, see 
https://docs.moodle.org/dev/Admin_settings#Individual_settings

**Comment by martignoni** [comment]
Many thanks for your work. As reviewed above, I think it's much better to implement the "buttons in footer" setting with Moodle API. I coded the configuration in this sense this afternoon. The remaining of your code is OK.

If you want to remove your "configuration setting" from your PR, I'd be glad to accept it, so that you'll be credited for this code. Otherwise, I'll close the PR and commit myself your changes: please tell me what you prefer.

**Comment by martignoni** [comment]
@adpe Do you have a time estimation about this? I'd like to have this done so I can close the issue.

**Comment by adpe** [comment]
> @adpe Do you have a time estimation about this? I'd like to have this done so I can close the issue.

I'll look to it this night, alright?

**Comment by martignoni** [comment]
Splendid!

**Comment by adpe** [comment]
> Brilliant. A big thank you.
> 
> I'm not convinced by your choice of the button's layout though. I'll see if I keep them :-).

Haha - I thought the same. Would be better to style both with the `btn btn-secondary` classes and maybe small?

Did you see the thing with the config value in the `settings.php`. I was first weird, because I was searching this field in the wrong table. It was stored in `prefix_config` and not in `prefix_config_plugins`.

Was glad to work on this.

**Comment by martignoni** [comment]
> Haha - I thought the same. Would be better to style both with the btn btn-secondary classes and maybe small?

Maybe, but for the moment, I think I'll keep them the default way. They could be much better, but there's a bug in Moodle ([MDL-58607](http://tracker.moodle.org/browse/MDL-58607)) that prevents correct styling, so it's maybe better to let them this way until this is fixed. I tried to send a fix, but could not convince that my fix was good enough.

> Did you see the thing with the config value in the settings.php. I was first weird, because I was searching this field in the wrong table. It was stored in prefix_config and not in prefix_config_plugins.

Yes, it was much better, so I adopted it. 👍

**Pull request merged**